src/spec.h

```diff
@@ -283,6 +283,9 @@ typedef struct IndexSpec {
 
   // Count the number of times the index was used
   long long counter;
+
+  // bitmap of dialects used by this index
+  unsigned long long dialects;
```

Comment:
I suggest lowering the size (no need for 64 "live" dialects) - will decrease the additional used memory for use cases with multiple indices.
(rename is less critical than size)
```suggestion
  unsigned long dialects_usage;
```

Reply:
Regarding length, sure we can change. For right now, all we need is an `unsigned char`. In `IndexSpec`, the size of the struct will be unchanged regardless of the size of the bitarray. Be it 8 bits or 64, the size of the struct will grow by 64 bits. I figured we might as well use all the available bits that we allocate for future-proofing. I personally don't like `long` in general, as that is different sizes on different arch (eg 32 bit on windows, 64 on linux), whereas long long is almost always 64 bits. In specific, I think `uint_leastN_t` would be preferable, with N being up for debate (8?).